### PR TITLE
Added support to auto- vet\tets\build on file change.

### PR DESCRIPTION
**Pull request by s-KaiNet**
solves #49  
**TL;DR;**  
run `gulp watch --dev` to automatically generate development package (unminified, with comments and sourcemaps) on file change. 

**Full version:**  
`watch` task setup watchers for source files, spec files and build files.  
When source file changes following process will start:  
`source change -> vet -> transpile ts -> test -> build with webpack`  
Tasks run one after another and if any fails, whole chain will be broken. This achieved by adding dependencies to existing tasks (transpile depends on vet, test depends on transpile (you need source .js files for tests), build depends on test (if test is failed build is not valid)). This also means that for successful build you need to fix all vet and test errors.    

When spec changes following will start:  
`spec change -> vet -> transpile ts -> test`  
In this case build package won't be creted, because source files are not changed, so run test only.   

When build file changed, only transpilation process will start (source and specs are not changed).   

`watch` task supports all params supported by other tasks - `debug`, `specs`, `verbose`, `version`. Params will be passed to appropriate tasks, but  
**CHANGE**: for `build-lib` `debug` param renamed to `dev`, because when running with `gulp watch --debug` it also affects `test` task and produces very long output, which may be undesirable.  

Looking for feedback @ngOfficeUIFabric/ngofficeuifabric-members

### CODE REVIEW

**Comment by andrewconnell** [comment]
Very well done... manually merged. Thanks!
